src/menu.tsx

```diff
@@ -1,18 +1,155 @@
+}
+
+type MenuItemProps = MenuItemWithControlProps | MenuItemWithoutControlProps;
+
+export const MenuItem: React.FC<MenuItemProps> = ({
+    label,
+    Icon,
+    destructive,
+    disabled,
+    'aria-label': ariaLabel,
+    onPress,
+    controlType,
+    checked,
+}) => {
+    const {focusedValue, setFocusedValue, closeMenu, isMenuOpen} = useMenuContext();
+    const contentColor = destructive ? vars.colors.textLinkDanger : vars.colors.neutralHigh;
+
+    const renderContent = () =>
+        controlType === 'checkbox' ? (
+            <Checkbox
+                name={label}
+                checked={checked}
+                onChange={() => {
+                    if (isMenuOpen) {
+                        onPress(label);
+                        closeMenu();
+                    }
+                }}
+                disabled={disabled}
+                role="menuitemcheckbox"
+                aria-label={ariaLabel ?? label}
+                render={({controlElement}) => (
+                    <Box paddingX={8} paddingY={12}>
+                        <Inline space="between" alignItems="center">
+                            <Inline space={8} alignItems="center">
+                                {Icon && <Icon size={24} color={contentColor} />}
+                                <Text3 regular color={contentColor}>
+                                    {label}
+                                </Text3>
+                            </Inline>
+                            <Box paddingLeft={16}>{controlElement}</Box>
+                        </Inline>
+                    </Box>
+                )}
+            />
+        ) : (
+            <Touchable
+                onPress={() => {
+                    if (isMenuOpen) {
+                        onPress(label);
+                        closeMenu();
+                    }
+                }}
+                disabled={disabled}
+                role="menuitem"
+                aria-label={label}
+            >
+                <Box paddingX={8} paddingY={12}>
+                    <Inline space={8} alignItems="center">
+                        {Icon && <Icon size={24} color={contentColor} />}
+                        <Text3 regular color={contentColor}>
+                            {label}
+                        </Text3>
+                    </Inline>
+                </Box>
+            </Touchable>
+        );
+
+    return (
+        <div
+            className={classnames(styles.menuItem, {
+                [styles.menuItemDisabled]: disabled,
+                [styles.menuItemHovered]: !disabled && focusedValue === label,
+            })}
+            onMouseMove={() => setFocusedValue(disabled ? null : label)}
+            onMouseLeave={() => setFocusedValue(null)}
+        >
+            {renderContent()}
+        </div>
+    );
+};
+
+type MenuSectionProps = {
+    children?: React.ReactNode;
+};
+
+export const MenuSection: React.FC<MenuSectionProps> = ({children}) => {
```

Comment:
perhaps we could skip the render when children is empty

Reply:
Added the null check. Anyways, I believe that scenario is a bit weird and also not a very good usage of the component.